src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -295,6 +295,21 @@ private void handleServiceStateChange(GreengrassService greengrassService, State
             updatedGreengrassServiceSet.add(greengrassService);
         }
 
+        // Always report status on errored state, evaluate overall status based on current state of all components
+        if (newState.equals(State.ERRORED)) {
+            synchronized (updatedGreengrassServiceSet) {
+                Instant now = Instant.now();
+                AtomicReference<OverallStatus> overAllStatus = new AtomicReference<>();
```

Comment:
we've seen some circumstances where overallStatus is null - is it worth initializing it to HEALTHY?

Reply:
overallStatus would be null if kernel just launches and there is no service running. Should we consider it healthy or assign a new value like "UNKNOWN"?

(I think we can address this issue in a separate pr.